File: frontend-angular/src/app/model/expense/expense.ts
export class addExpense {
    id?: number
    userId?: number
    amount: number
    category: string
    date: Date

    constructor() {
        this.id = 0;
        this.userId = 0;
        this.amount = 0;
        this.category = ""
        this.date = new Date()
    }
}

export interface GetExpense {
    id: number
    amount: number
    category: string
    date: Date
    userId: number
}

File: backend-service-springboot/src/main/java/com/sadman/financial/entity/Expense.java
package com.sadman.financial.entity;

import com.sadman.financial.enums.ExpenseCategory;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Data
@Entity
@Table(name = "expense")
public class Expense extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Double amount;

    @Enumerated(EnumType.STRING)
    private ExpenseCategory category;

    @OneToMany(mappedBy = "expense", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Contract> contracts;

    private LocalDate date;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

}